### DISCUSSION TITLE
Issue with data retrival Error: getaddrinfo ENOTFOUND

### DISCUSSION THREAD

**Discussion by kkggg55** [supabase/supabase#20934] [Questions]
I'm encountering an issue where my APIs are unable to fetch data from the database. Specifically, the error I'm receiving is: **Error: getaddrinfo ENOTFOUND db.database_url.supabase.co**. This issue came up unexpectedly as there have been no recent changes to the codebase, and to the best of my knowledge, the database is not corrupted.

Here's a brief overview of the tech stack and relevant frameworks used in the project:

1. Next.js
2. React
3. NPM (Node Package Manager)
4. Database interaction tools: Prisma, GraphQL, Kysely

Could you provide guidance or suggestions on what might be causing this ENOTFOUND error and how to resolve it? Any insights into potential issues with the DNS resolution or other networking concerns related to db.database_url.supabase.co would be greatly appreciated.

**Comment by GaryAustin1** [comment]
https://github.com/orgs/supabase/discussions/17817
Most likely do to ipv4 to ipv6 transition.  If the issue is with Prisma.

**Reply by kkggg55** [reply]
will try this

**Reply by kkggg55** [reply]
I tried switching to supavisor via this url postgres://postgres.lxrbpmzjqdsvvgvwvgep:[YOUR-PASSWORD]@aws-0-us-west-1.pooler.supabase.com:6543/postgres but our app could still not connect to the database

**Reply by Megahedron69** [reply]
Have you found a solution?

**Reply by khaelou** [reply]
In my case, I was getting this error when migrating a Vue 3 Firebase project to Supabase with the migration tools. I simply changed db.XXXX.supabase.co to aws-XXXX-pooler-.supabase.com in supabase-service.json and the error left 🙂

**Comment by InsightfulFuture** [comment]
I am having the same issue, but it also says it can't connect in the supabase dashboard & I am using the supabase client library. Any updates?

**Comment by Megahedron69** [comment]
I just change my dns whenever I work on the project (to cloudfare 1.1.1.1)
and connection gets established but I have also not found any permanent
solution

Regards

**Comment by joshberc** [comment]
Instead of using a direct connection string, use the Transaction or Session pooler connection string, resolved this issue for me. Using Drizzle to connect to a postgres DB on Supabase.

**Reply by cxhello** [reply]
Use resolved Session pooler connection string thanks. @joshberc

**Reply by 100stacks** [reply]
Thanks @joshberc! 🙌  This also worked for me.  

Error while running a Drizzle migration:

```node
⏳ Running migrations...
❌ Migration failed
Error: getaddrinfo ENOTFOUND db.dzghyjxehylpnttutwzq.supabase.co
    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:120:26) {
  errno: -3008,
  code: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'db.dzghyjxehylpnttutwzq.supabase.co'
}
```

Stack:

- Next.js@14.2.24
- Drizzle ORM@0.31.2
- Supabase
- AWS CDK@2.178.0
- AWS Lambda@2.178.0

**Reply by kleenkanteen** [reply]
no work for me

**Reply by therajusah** [reply]
Worked, Thanks. I am deploying node.js backend on vercel.

**Reply by tahaturkay** [reply]
It worked!!! Thank you so much 🙏

**Comment by shaffatshah** [comment]
It actually says in my Supabase db setttings area that **Pool mode is permanently set to Transaction on port 6543**
_You can use Session mode by connecting to the pooler on port 5432 instead_ and chosing 'Transaction' worked for me

**Comment by jobissjo** [comment]
## Issue with with executing 

```python script
SQLALCHEMY_DATABASE_URL = f"postgresql+asyncpg://{PG_USER}:{PG_PASSWORD}@{PG_HOST}:{PG_PORT}/{PG_DB}"

engine = create_async_engine(
    config.SQLALCHEMY_DATABASE_URL,
    echo=False,
)  

async_session_maker = async_sessionmaker(engine, expire_on_commit=False)

async def get_async_db_session() -> AsyncGenerator[AsyncSession, None]: 
    async with async_session_maker() as session:
        yield session


async def register_user(user_data:user_schema.RegisterSchema,
            db: AsyncSession):
        query = select(User).where(
            (User.username == user_data.username) |
            (User.email == user_data.email)
        )
        result = await db.execute(query)

```
        
      

[Errno 11003] getaddrinfo failed


I am getting this error, anyone know how to fix

**Comment by braden-w** [comment]
This was the solution for me:

If you're using an ORM like Drizzle or Prisma with Supabase, you actually use a different connection string than the one shown on the front page of the transaction pooler.

The front page will show something like:
```plaintext
postgres://postgres:[YOUR-PASSWORD]@db.[YOUR-PROJECT-ID].supabase.co:6543/postgres
```
![CleanShot 2025-04-20 at 17 21 03](https://github.com/user-attachments/assets/be607c69-5e0b-436e-a55b-644d63f34a23)


While the ORM page (in Database Settings) will show:
```plaintext
postgresql://postgres.[YOUR-PROJECT-ID]:[YOUR-PASSWORD]@aws-0-us-east-2.pooler.supabase.com:6543/postgres
```

![CleanShot 2025-04-20 at 17 21 15](https://github.com/user-attachments/assets/05559f81-a80d-43f3-b4f8-43cd555f71c1)


There are a few key differences between these URLs:
- The protocol changes from `postgres://` to `postgresql://`
- The username includes your project ref (`postgres.totssvlmppdukrrmphsg` instead of just `postgres`)
- The host uses an AWS-based URL instead of the standard Supabase one

This actually explains why @khaelou's solution of switching to an AWS-based URL worked - the ORM connection string needs to use these AWS endpoints directly. Hope this helps anyone else running into the same issue!

**Reply by Lermatroid** [reply]
Is this a fix though? Certain things like [zero sync](https://zero.rocicorp.dev/docs/connecting-to-postgres#supabase) require the direct connection so using it via the pooler is not always an option.

**Reply by tranhnh** [reply]
thxxxxxxx goat!

**Reply by Sarp2** [reply]
Thank you so much braden!

**Comment by Fayiz-A** [comment]
In my case, the problem was with the network. I changed from wifi to mobile data hotspot, and it worked.
